### ISSUE TITLE
[BUG] tailwindcss/migration-from-tailwind-2 rule issues

### ISSUE THREAD

**Issue by dbtek** [francoismassart/eslint-plugin-tailwindcss#128]
**Describe the bug**
Using `bg-opacity-*` or `ring-opacity-*` classes results in lint errors.

**To Reproduce**
1. Use one of the classnames `bg-opacity-10, ring-opacity-5`
2. See error

**Expected behavior**
Allow correct usage of opacity classes.

**Screenshots**
<img width="1152" alt="image" src="https://user-images.githubusercontent.com/310612/155888645-62d9a5fc-4f97-48d8-a25a-5d539317066d.png">

**Environment (please complete the following information):**
 - OS: macOs
 - Softwares + version used:
     -  node 16, eslint 7.32.0, eslint-plugin-tailwindcss 7.28.0

**Additional context**
None

**eslint config file or live demo**
```js
module.exports = {
  ignorePatterns: [
    '.eslintrc.js',
    'gatsby-config.js',
    'gatsby-node.js',
    'src/graphql/graphqlTypes.ts',
  ],
  env: {
    browser: true,
    es2021: true,
  },
  extends: [
    'eslint:recommended',
    'plugin:react/recommended',
    'plugin:react-hooks/recommended',
    'plugin:@typescript-eslint/recommended',
    'plugin:react/jsx-runtime',
    'plugin:tailwindcss/recommended',
    'plugin:prettier/recommended',
  ],
  settings: {
    react: {
      version: 'detect',
    },
  },
  parser: '@typescript-eslint/parser',
  parserOptions: {
    ecmaFeatures: {
      jsx: true,
    },
    ecmaVersion: 12,
  },
  plugins: ['react', '@typescript-eslint', 'tailwindcss'],
  rules: {
    'react/jsx-uses-react': 'error',
    'react/jsx-uses-vars': 'error',
    'react/prop-types': 0,
  },
};

```

**Comment by francoismassart** [comment]
Hi @dbtek & @charkour, 👋 
This is on purpose.

If you take a look 👀  at the [Tailwind CSS docs for v3.x.x](https://tailwindcss.com/docs), you will not find any page about:

- `bg-opacity`
- `ring-opacity`
- ...

If you read [background-color docs about opacity](https://tailwindcss.com/docs/background-color#changing-the-opacity) you can see that they recommend using the **opacity suffix** (e.g. `/50`) like so 
`<button class="bg-sky-500/50 ..."></button>`.

As they explain in the [upgrade guide](https://tailwindcss.com/docs/upgrade-guide#class-name-changes):
> Wherever possible we have preserved the old name as well so many of these changes are non-breaking, but you’re encouraged to update to the new class names.

That's what these rules are meant for: **encouraging you to update to the new classnames** 😉 

So, this is not a bug, it is made on purpose.

▶️ If you don't like it, you can easily disable the rule in your eslintrc file.

Hope it helps.

**Comment by dbtek** [comment]
Sorry, that's my bad.

@francoismassart thanks for heads up. Used components from Tailwind UI, didn't think they're outdated.

**Comment by charkour** [comment]
Thanks for the quick response. I had also misread that. Thank you.

**Comment by vedmant** [comment]
bg-opacity is not outdated, it's available and active in Tailwind 3 https://tailwindcss.com/docs/opacity
Also it's useful when using components with bg color and adding opacity conditionally.
